flink-connectors/flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSGeneralUtil.java

```diff
@@ -369,4 +367,35 @@ public static void closeResources(SdkAutoCloseable... resources) {
             throw exception;
         }
     }
+
+    public static void validateWebIdentityTokenFileCredentialsProvider(Properties config) {
+        validateCredentialProvider(config);
+        try {
+            CredentialProvider credentialProviderType =
+                    getCredentialProviderType(config, AWSConfigConstants.AWS_CREDENTIALS_PROVIDER);
+            if (credentialProviderType.equals(WEB_IDENTITY_TOKEN)) {
+                getCredentialsProvider(config).resolveCredentials();
```

Comment:
Where is this code executed? Is it on client or Task Manager? If client, then client will need AWS creds, this is not idea if job is being submitted from a host will limited permissions

Reply:
This is executed on the TM by the sink writer, the web Identity token falls back to env variables so verifying any missing credentials is not possible by verifying the configs only. I thought it would be more thorough to catch any failure on credential resolving and report it .
I believe the code is going to be re-executed by the client when sending requests using same credentials later, why should a failure here be ignored or not an indicator of bad credentials?

Reply:
If this code runs on the TM it is ok. I just wanted to avoid the situation where the job fails to start because client has insufficient permissions